src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -453,7 +453,8 @@ private String DecodeEntryName(Byte[] entryNameBytes)
                 readEntryNameEncoding = Encoding.UTF8;
             }
 
-            return new string(readEntryNameEncoding.GetChars(entryNameBytes)); //readEntryNameEncoding.GetString(entryNameBytes);
+            fixed (byte* bytePtr = entryNameBytes)
+                return readEntryNameEncoding.GetString(bytePtr, entryNameBytes.Length);
```

Comment:
While I can appreciate that argument, all of the important Encoding-derived types override that GetString overload.  It'd be extremely rare to end up in a situation here where an encoding was being used that didn't, and as such I'd much rather avoid unsafe code.